### PR TITLE
Use Apache HTTP Client

### PR DESCRIPTION
**Pull request by ctienshi**
## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? yes
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

### CODE REVIEW

File: src/main/java/org/wso2/callhome/internal/CallHome.java [hunk 1]
```diff
@@ -300,52 +323,53 @@ private URL constructCallHomeURL(ExtractedInfo extractedInfo) throws CallHomeExc
     }
 
     /**
-     * This method creates an HTTPS connection to a given url.
+     * This method creates a ClosableHTTPClient with the WSO2 TrustStore as a trust manager.
```
Comment (kasunsiyambalapitiya) [inline src/main/java/org/wso2/callhome/internal/CallHome.java]: Shall we correct the comment as follows
`TrustStore as a trust manager` -> `TrustStore as the trust manager`

File: src/main/java/org/wso2/callhome/internal/CallHome.java [hunk 2]
```diff
@@ -300,52 +323,53 @@ private URL constructCallHomeURL(ExtractedInfo extractedInfo) throws CallHomeExc
...
-                        response.append(readLine);
-                    }
-                }
-            }
-        } catch (IOException e) {
-            log.debug("Error while setting request method " + e.getMessage());
-            throw new CallHomeException("Error while setting request method", e);
+            return responseHandler.handleResponse(response);
+        } catch (
+                IOException e) {
```
Comment (kasunsiyambalapitiya) [inline src/main/java/org/wso2/callhome/internal/CallHome.java]: Shall we make this inline with `catch(... )` ?

File: src/main/java/org/wso2/callhome/internal/CallHomeComponent.java [hunk 3]
```diff
@@ -92,11 +63,31 @@ public void deactivate() {
...
+            service = ConfigProvider.class,
+            cardinality = ReferenceCardinality.AT_LEAST_ONE,
+            policy = ReferencePolicy.DYNAMIC,
+            unbind = "unregisterConfigProviderService"
+    )
+    protected void registerConfigProviderService(ConfigProvider configProvider) {
+
+        logger.debug("ConfigProvider service registered");
+        DataHolder.getInstance().setConfigProvider(configProvider);
+
```
Comment (kasunsiyambalapitiya) [inline src/main/java/org/wso2/callhome/internal/CallHomeComponent.java]: Shall we remove the additional empty line?

Comment (ctienshi) [reply src/main/java/org/wso2/callhome/internal/CallHomeComponent.java]: fixed

File: src/main/java/org/wso2/callhome/internal/CallHome.java (hunk 2 above)
Comment (ctienshi) [reply src/main/java/org/wso2/callhome/internal/CallHome.java]: fixed

File: src/main/java/org/wso2/callhome/internal/CallHome.java (hunk 1 above)
Comment (ctienshi) [reply src/main/java/org/wso2/callhome/internal/CallHome.java]: fixed

**Pull request merged**